optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/statistic/StatisticType.java

```diff
@@ -23,4 +23,9 @@
      */
     public String name();
 
+    /**
+     * @return never null
+     */
+    public String formattedName();
```

Comment:
Good idea to pretty up the report, just some thoughts on naming, to be consistent with the rest of the code:

Proposal A) Either we use "getLabel()" instead of "getFormattedName()" (and rename name() to getName())
This is in sync with the example domain objects which often have a toString() for logging and getLabel() for the UI (but no getName();

Proposal B) "getFormattedName()" and "name()" both becomes "getName()". A StatisticType gets a field name and a constructor that sets it. This is similar to JobType.
The filename for the csv stats is the getName(), just like the solver benchmark name and the problem benchmark name are used for the directory names.
Advantages:
- name can differ from enum key. For example: "Semi-special stat" and SEMI_SPECIAL_STAT.
- better localization opportunity